src/movies.js

```diff
@@ -1,35 +1,79 @@
 // The `movies` array from the file `src/data.js`.
-console.log('movies: ', movies);
-
+//console.log('movies: ', movies);
 
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(moviesArr) {
+  const moviesDirec = moviesArr.map((movie) => {
+    return movie.director;
+  });
+  return moviesDirec;
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(moviesArr) {
+  const stevenDramas = moviesArr.filter((movie) => {
+    return (
+      movie.director === 'Steven Spielberg' && movie.genre.includes('Drama')
+    );
+  });
+  return stevenDramas.length;
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(moviesArr) {
+  if (!moviesArr.length) return 0;
+  let totalScore = moviesArr.reduce((acc, val) => {
+    if (!val.score) {
+      return acc;
+    }
+    return acc + val.score;
+  }, 0);
+  let avg = totalScore / moviesArr.length;
+  let dec = Number(avg.toFixed(2));
+  return dec;
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore() {}
+function dramaMoviesScore(moviesArr) {
+  const dramaMoviesAvg = moviesArr.filter((movie) =>
+    movie.genre.includes('Drama')
+  );
+  return scoresAverage(dramaMoviesAvg);
+}
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear() {}
+function orderByYear(movieArr) {
+  let year = [...movieArr].sort((a, b) => {
+    if (a.year === b.year) {
+      return a.title.localeCompare(b.title);
+    } else {
+      return a.year - b.year;
+    }
+  });
+  return year;
+}
 
 // Iteration 6: Alphabetic Order - Order by title and print the first 20 titles
-function orderAlphabetically() {}
+function orderAlphabetically(moviesArr) {
+  let moviesTitles = moviesArr.map((movie) =>  movie.title);
+
+  let orderedTitles = moviesTitles.sort((a, b) => a.localeCompare(b));
+  if (orderedTitles.length > 20) {
+    return orderedTitles.slice(0, 20);
+  } else if (orderedTitles.length < 20) {
+    return orderedTitles.slice(0, 20);
+  }
```

Comment:
No need for the if statement, if you see, you are writing 2 different conditions with the exact same return